<a href="https://colab.research.google.com/github/Debdatta7/PrepInsta-Internship/blob/main/SQL_murder_mystery_sol_Debdatta_Mukherjee.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### SQL Murder Mystery



In [ ]:
import pandas as pd
import sqlite3 as sql

In [ ]:
con = sql.connect("/content/drive/MyDrive/Data analytics project/Week5/sql-murder-mystery.db")

#### Retrieve Crime Scene Report:

In [ ]:
query = """
SELECT *
FROM crime_scene_report
WHERE City = "SQL City"
ORDER BY date;
"""
pd.set_option('display.max_colwidth',None)
pd.read_sql_query(query,con)

date     type  \
0  20170712    theft   
1  20170820    arson   
2  20171110  robbery   
3  20180103  bribery   
4  20180115  assault   
5  20180115  assault   
6  20180115   murder   
7  20180215   murder   
8  20180215   murder   

                                                                                                                                                                                                                                                              description  \
0              A lone hunter stalks the night, firing arrows into the Darkness.\n            There is no hiding, no escape. In the distance, the beast\n            falters, tethered to the void. The killing blow comes without\n            hesitation, without mercy.   
1                                                                                                                                                                  Wield the Hammer of Sol with honor, Titan, it is a thing of\n            legend, both past and future.   
2  The Gjallarhorn shoulder-mounted rocket system was forged from\n            the armor of Guardians who fell at the Twilight Gap. Gifted\n            to the survivors of that terrible battle, the Gjallarhorn\n            is seen as a symbol of honor and survival.   
3                                                                                                                                                 Apparently, Cayde thought it necessary to expose this extremely\n            rare vegetable to a Hive summoning ritual.   
4                                                                                                                                                                                             Hamilton: Lee, do you yield? Burr: You shot him in the side! Yes he yields!   
5                                                                                                                                                                                                                                                        Report Not Found   
6                                                                               Security footage shows that there were 2 witnesses. The first witness lives at the last house on "Northwestern Dr". The second witness, named Annabel, lives somewhere on "Franklin Ave".   
7                                                                                                                                                                                                                                              REDACTED REDACTED REDACTED   
8                                                                                                                                                                                                                 Someone killed the guard! He took an arrow to the knee!   

       city  
0  SQL City  
1  SQL City  
2  SQL City  
3  SQL City  
4  SQL City  
5  SQL City  
6  SQL City  
7  SQL City  
8  SQL City

#### Witness Personal Details:

In [ ]:
# Witness 1

query = """
SELECT *
FROM person
WHERE address_street_name = 'Northwestern Dr'
ORDER BY address_number;
"""
pd.read_sql_query(query,con)

id                  name  license_id  address_number  \
0   89906       Kinsey Erickson      510019             309   
1   15171          Weldon Penso      336999             311   
2   40336       Fredia Billeter      251082             404   
3   85280             Donn Duin      569122             540   
4   12711        Norman Apolito      667757             599   
5   75484          Del Tacderen      447887             607   
6   26758      Angelena Billman      942074             629   
7   39688            Abe Roeker      537445             689   
8   10010             Muoi Cary      385336             741   
9   40760          Renata Faver      976650             745   
10  25615         Olevia Morena      388477             812   
11  99826          Ivy Kazarian      994470             892   
12  44542        Ivory Mcgrevey      873842             897   
13  55130        Shiloh Huntoon      472118             951   
14  33415         Homer Simpson      108392             981   
15  29608       Willetta Lurvey      800568            1028   
16  18376              Josh Shi      653712            1091   
17  22239         Dusty Sigafus      710517            1125   
18  70485     Andera Kostelecky      354822            1252   
19  23960  Kristopher Lagerberg      658777            1392   
20  71924        Laurie Scronce      774116            1418   
21  81052     Calvin Czelusniak      471483            1516   
22  72076     Courtney Bordeaux      324614            1707   
23  51552           Bonita Knie      764124            1726   
24  88423           Teri Ehrich      510876            1879   
25  81885         Kurtis Masuda      934201            2108   
26  96178           Galina Khuu      450248            2183   
27  80921       Donovan Yackley      222395            2386   
28  34352    Vincenza Burkhardt      436470            2512   
29  53699      Christian Pagano      628433            2585   
30  54564         Sammie Sengun      217286            2590   
31  90397     Dionna Kranwinkle      797260            2678   
32  34155      Allyson Lazenson      814209            2840   
33  59307          Cyril Yongue      753524            2903   
34  98593           Jonah Toner      578982            2947   
35  44004           Alison Eska      327797            2951   
36  68690            Yer Modest      852404            3046   
37  51114     Christena Saffell      814284            3055   
38  23044          Val Portlock      924989            3143   
39  28360        Rashad Cascone      924517            3212   
40  53076           Boris Bijou      664914            3327   
41  36378        Freddie Ellzey      267882            3449   
42  87456   Leonora Wolfsberger      215868            3483   
43  93509        Emmitt Aceuedo      916706            3491   
44  19420           Cody Schiel      890431            3524   
45  96595         Coretta Cubie      303645            3631   
46  73368        Torie Thalmann      773862            3697   
47  53890        Sophie Tiberio      957671            3755   
48  17729        Lasonya Wildey      439686            3824   
49  14887        Morty Schapiro      118009            4919   

   address_street_name        ssn  
0      Northwestern Dr  635287661  
1      Northwestern Dr  131379495  
2      Northwestern Dr  108056488  
3      Northwestern Dr  613873810  
4      Northwestern Dr  778264744  
5      Northwestern Dr  243821337  
6      Northwestern Dr  341854756  
7      Northwestern Dr  438969447  
8      Northwestern Dr  828638512  
9      Northwestern Dr  547572040  
10     Northwestern Dr  962831331  
11     Northwestern Dr  208359813  
12     Northwestern Dr  988557407  
13     Northwestern Dr  826647644  
14     Northwestern Dr  871392019  
15     Northwestern Dr  843751378  
16     Northwestern Dr  193899001  
17     Northwestern Dr  724386723  
18     Northwestern Dr  787423012  
19     Northwestern Dr  492912529  
20     Northwestern Dr  254883636  
2

In [ ]:
# Witness 2

query = """
SELECT *
FROM person
WHERE address_street_name = 'Franklin Ave'
AND name LIKE '%Annabel%';
"""
pd.read_sql_query(query,con)

id            name  license_id  address_number address_street_name  \
0  16371  Annabel Miller      490173             103        Franklin Ave   

         ssn  
0  318771143

#### View Witness Interviews:

In [ ]:
# Witness 1

query = """
SELECT *
FROM interview
WHERE person_id = '16371';
"""
pd.read_sql_query(query,con)

person_id  \
0      16371   

                                                                                                              transcript  
0  I saw the murder happen, and I recognized the killer from my gym when I was working out last week on January the 9th.

In [ ]:
# Witness 2

query = """
SELECT *
FROM interview
WHERE person_id = '14887';
"""
pd.read_sql_query(query,con)

person_id  \
0      14887   

                                                                                                                                                                                                                        transcript  
0  I heard a gunshot and then saw a man run out. He had a "Get Fit Now Gym" bag. The membership number on the bag started with "48Z". Only gold members have those bags. The man got into a car with a plate that included "H42W".

#### Check Gym Database:

In [ ]:
query = """
SELECT *
FROM get_fit_now_check_in
WHERE check_in_date = '20180109'
AND membership_id LIKE '48Z%';
"""
pd.read_sql_query(query,con)

membership_id  check_in_date  check_in_time  check_out_time
0         48Z7A       20180109           1600            1730
1         48Z55       20180109           1530            1700

#### Check Car Details:

In [ ]:
query = """
SELECT *
FROM drivers_license
WHERE plate_number LIKE '%H42W%';
"""
pd.read_sql_query(query,con)

id  age  height eye_color hair_color  gender plate_number   car_make  \
0  183779   21      65      blue     blonde  female       H42W0X     Toyota   
1  423327   30      70     brown      brown    male       0H42W2  Chevrolet   
2  664760   21      71     black      black    male       4H42WR     Nissan   

  car_model  
0     Prius  
1  Spark LS  
2    Altima

#### Personal Details:

In [ ]:
# Witness 1

query = """
SELECT *
FROM person
WHERE license_id = '423327';
"""
pd.read_sql_query(query,con)

id           name  license_id  address_number    address_street_name  \
0  67318  Jeremy Bowers      423327             530  Washington Pl, Apt 3A   

         ssn  
0  871539279

In [ ]:
# Witness 2

query = """
SELECT *
FROM person
WHERE license_id = '664760';
"""
pd.read_sql_query(query,con)

id            name  license_id  address_number address_street_name  \
0  51739  Tushar Chandra      664760             312              Phi St   

         ssn  
0  137882671

#### Membership Status at the Gym:

In [ ]:
query = """
SELECT *
FROM get_fit_now_member
WHERE membership_status = 'gold'
AND id LIKE '48Z%';
"""
pd.read_sql_query(query,con)

id  person_id           name  membership_start_date membership_status
0  48Z7A      28819   Joe Germuska               20160305              gold
1  48Z55      67318  Jeremy Bowers               20160101              gold

####  I N V E S T I G A T I O N    R E P O R T

Jeremy Bowers is the murderer!!!

 Because of the following reasons:
 1. From the crime scene report we can find that we have two witnesses,
  1. One who lives at the last house on "Northwestern Dr"
	2. And the other whose name is "Annabel" and lives somewhere on "Franklin Ave"

2. (a) Our first witness 'Annabel' said that she saw the murderer last week on her gym on 09.01.2018,  
  (b) Our second witness 'Morty Schapiro' said that:
   1.	He saw a man running out, after hearing a gunshot.
   2.	He had a "Get Fit Now Gym" bag
   3. The membership number started with "48Z" - Only "Gold Members" have that kind of bag.
   4.	The man got into a car whose number plate included the following initials - "H42W"

3. After getting the car details, I found out that there were total 3 persons (1 Female and 2 Males) with the number plate including "H42W" in it.
  1. Suspect 1 : Jeremy Bowers/ ID: 67318 <--<
  2. Suspect 2 : Tushar Chandra/ ID: 51739

4. After reviewing the membership status of the gym, I found out that there were 2 persons with the "Gold Membership" and "48Z" initials on their gym bag.
  1. Suspect 1 : Joe Germuska
  2. Suspect 2 : Jeremy Bowers <--<

5. After analysing,
	 I found out that the murderer was 'Jeremy Bowers' who was a gold member at the gym, with bag no: "48Z55" who murdered on 15.01.2018, and
	 then fled with his car having number plate "0H42W2".